### PR TITLE
Feature/stream

### PR DESCRIPTION
**Pull request by adamegyed**
Implemented streaming by adding a new "PutStream" output to the KVStore, then using a func function as a passthrough to the command line tool (with input validation, unwrapping, etc).

### CODE REVIEW

File: func/func_server.cpp [hunk 1]
```diff
@@ -25,6 +26,8 @@ Status FuncServiceImpl::hook(ServerContext *context, const HookRequest *request,
     function_map_[event_func] = warble_funcs::Read;
   } else if (event_func == "Profile") {
     function_map_[event_func] = warble_funcs::Profile;
+  } else if (event_func == "Stream") {
+    stream_function_map_[event_func] = warble_funcs::Stream;
```
Comment (etxyc) [inline func/func_server.cpp:30]: Unhook "stream" is also supposed to be handle.

**Comment by etxyc** [comment]
Overall a nice and efficient design. One thing could possibly improve is that storing kv pairs for all stream request is not space-efficient. Just ID or key is needed to be stored in the queue since key is unique in kvstore and is much smaller than value. This could save a lot of space when the project scales.

File: func/func_server.cpp (hunk 1 above)
Comment (adamegyed) [reply func/func_server.cpp:30]: Good catch. I've just updated the server to include it.

Comment (etxyc) [reply func/func_server.cpp:30]: Good to merge

**Pull request merged**